2_Scope_Closures/practice_new/closure_part3_gim.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const INPUT_LENGTH = 1;
+const INPUT_INVALID_MSG = 'this is invalid input.';
+const BASE = 10;
+
+function calculator() {
+  const operators = ['+', '-', '*', '/', '='];
+  let operator;
+  let number = 0;
+  let total = 0;
+
+  return calculate;
+
+  function calculate(input) {
+    if (!isValidInput(input)) {
+      throw new Error(INPUT_INVALID_MSG);
+    }
+    if (isOperator(input) && operate()) {
+      operator = input;
+      number = 0;
+
+      return operator === '=' ? formatTotal(total) : input;
+    }
+    if (operator === '=') {
+      operator = null;
```

Comment:
```suggestion
      operator = '';
```
null과 '' 모두 다 써도 문제가 없다면 null보다는 ''가 코드를 파악하는데 있어서 더 좋지 않을까해서 리뷰 남깁니다. 실제로 상단에 ```let operator;```에서 초기화를 안해주었는데,
물론 ```number = operator === '=' ? Number(input) : number * BASE + Number(input);```에서 처리를 해주었지만
 처음 읽었을 때 ```let operator = '';```로 초기화를 해주었다면 string에 관한 변수구나하고 조금 더 정보를 얻은 상태에서 코드를 읽어가서 가독성이 높아질 것 같습니다.

Reply:
동의합니다 null 처리 같은 경우엔 이번 파트에서 더이상 필요없는 변수에 대한 적절한 GC를 위해 null 처리 해주는 걸 보고 의식해서 쓴 거였는데 맨 처음 초기화 부분에서는 확실히 '' 가 필요할 것 같네요 ! 감사합니다 ㅎㅎ

Reply:
GC는 생각하지 못했는데 말해주셔서 앞으로 조금 더 신경쓰면서 코드를 짤 수 있을것 같아요! 감사합니다 ㅎㅎ

Reply:
초기화에 대한 얘기를 적으려고 했는데 이미 호찬님이 적으셨군요!! 

저도 GC를 고려하면서 프로그램을 작성하지 않았는데 좋은 지적이네요 감사합니다!👍👍

Reply:
네 ㅎㅎ 저도 다시 생각해 보니 해당 변수는 가변적으로 메모리를 많이 차지할 우려가 있는 객체 타입이 아니니 굳이 GC까지는 안해줘도 될 것 같았네요 ! 그냥 배운 내용 상기 차원에서 쓴 걸로 봐주시면 감사하겠습니다